src/database.cpp

```diff
@@ -184,6 +192,25 @@ std::shared_ptr<Database> DatabaseFactory::CreateDatabase(Config& config, bool l
 	if (logConnection)
 		Console::Out("Connecting to database (%s)...", dbdesc.c_str());
 
+	try
+	{
+		return std::shared_ptr<Database>(new Database(engine, dbHost, dbPort, dbUser, dbPass, dbName));
```

Comment:
Just in case you don’t know about the underlying allocs: Calling `new` performs a heap allocation as does instantiating a non-empty `shared_ptr`. Use of `make_shared` would accomplish the same goal with only a single heap alloc.

Reply:
TIL. I thought it was just a convenience function, I didn't realize there was a difference in the underlying allocations

Reply:
Even if you look at documentation, it’s not very clear. Had to double check myself before submitting comment yesterday to confirm.